src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -134,6 +136,99 @@ def ensure_app_roles_is_a_set(
         return application_roles
 
 
+class ReleaseChannel(dict[str, Any]):
```

Comment:
Maybe specify the exact dict structure we're expecting with a `TypedDict`?
It's the same type that `show_release_channels()` returns, right? We can reuse it there too.
That way we would also be able to remove the `type: ignore` comments below.